src/lib/functions/runtimes/js/index.ts

```diff
@@ -54,15 +66,23 @@ export const getBuildFunction = async ({ config, directory, errorExit, func, pro
   const functionDirectory = dirname(func.mainFile)
   const srcFiles = functionDirectory === directory ? [func.mainFile] : [functionDirectory]
 
-  // @ts-expect-error TS(2532) FIXME: Object is possibly 'undefined'.
-  return () => ({ schedule: metadata.schedule, srcFiles })
+  // eslint-disable-next-line @typescript-eslint/require-await -- Must be async to match the interface
+  const build: BuildFunction<JsBuildResult> = async () => ({ schedule: metadata?.schedule, srcFiles })
```

Comment:
@ndhoule I gotchu:
```suggestion
  const build: BuildFunction<JsBuildResult> = () => Promise.resolve({ schedule: metadata?.schedule, srcFiles })
```

(I don't like it, but whatever)